DotNetHowToMultipleDataSources/Program.cs

```diff
@@ -0,0 +1,153 @@
+
+            // The JSON sample data set has a collection name of "hotels"
+            // The JSON sample data set uses Cosmos DB change tracking for change detection
+            // This means that when the Cosmos DB indexer runs, it will be able to detect which data has
+            // changed since the last run using built in change tracking
+            // See this link for more information
+            // https://docs.microsoft.com/azure/search/search-howto-index-cosmosdb#indexing-changed-documents
+            DataSource cosmosDbDataSource = DataSource.DocumentDb(
+                name: "cosmos-db",
+                documentDbConnectionString: configuration["CosmosDBConnectionString"],
+                collectionName: "hotels",
+                useChangeDetection: true);
+            // The Cosmos DB data source does not need to be deleted if it was already created,
+            // but the connection string may need to be updated if it was changed
+            await searchService.DataSources.CreateOrUpdateAsync(cosmosDbDataSource);
+
+            Console.WriteLine("Creating Azure SQL indexer...");
+            Indexer sqlIndexer = new Indexer(
+                name: "azure-sql-indexer",
+                dataSourceName: sqlDataSource.Name,
+                targetIndexName: index.Name,
+                schedule: new IndexingSchedule(TimeSpan.FromDays(1)));
+            // Indexers contain metadata about how much they have already indexed
+            // If we already ran the sample, the indexer will remember that it already
+            // indexed the sample data and not run again
+            // To avoid this, reset the indexer if it exists
+            exists = await searchService.Indexers.ExistsAsync(sqlIndexer.Name);
+            if (exists)
+            {
+                await searchService.Indexers.ResetAsync(sqlIndexer.Name);
+            }
+
+            await searchService.Indexers.CreateOrUpdateAsync(sqlIndexer);
+
+            Console.WriteLine("Creating Cosmos DB indexer...");
+            Indexer cosmosDbIndexer = new Indexer(
+                name: "cosmos-db-indexer",
+                dataSourceName: cosmosDbDataSource.Name,
+                targetIndexName: index.Name,
+                schedule: new IndexingSchedule(TimeSpan.FromDays(1)));
+
+            // Indexers contain metadata about how much they have already indexed
+            // If we already ran the sample, the indexer will remember that it already
+            // indexed the sample data and not run again
+            // To avoid this, reset the indexer if it exists
+            exists = await searchService.Indexers.ExistsAsync(cosmosDbIndexer.Name);
+            if (exists)
+            {
+                await searchService.Indexers.ResetAsync(cosmosDbIndexer.Name);
+            }
+
+            await searchService.Indexers.CreateOrUpdateAsync(cosmosDbIndexer);
+
+            // We created two indexer with schedules, but we also
```

Comment:
@mgottein Why bother with schedules then?

Reply:
Don't you need a schedule for "run once"? I thought it was a type of schedule.  Regardless, people copy-paste samples for their own code.  I think its useful to have schedule info and let people remove whatever they don't need.

Reply:
@brjohnstmsft we want to always create schedules, we never recommend creating an indexer without a schedule. This is due to transient conditions that can cause the indexer to fail